roles/maas_nameserver/defaults/main.yml

```diff
@@ -0,0 +1,11 @@
+---
+maas_api_url: "http://localhost:5240/MAAS/api/2.0/"  # Default, overridden by secrets
+maas_profile: "admin"  # Default, overridden by secrets
+default_domains:
+  - "maas"
+
+# List of target hosts for DNS records, excluding the 'maas' group by default.
+# Can be overridden in secrets/maas.yml (e.g., target_hosts: ['host1.example.com', 'host2.example.com'])
+# or via extra vars (e.g., ansible-playbook -e "target_hosts=['host1.example.com']").
+# If not set, defaults to all hosts in inventory groups except 'maas'.
+target_hosts: []
```

Comment:
Is this an ansible group that contains other ansible inventory groups?  Or is this saying "Only update DNS records for hosts in `target_hosts`?

Reply:
target_hosts will just contain all hosts from inventory groups except 'maas', it's not an ansible_group

Reply:
Is there a way to exclude certain ansible groups?  We may not want DNS records for all entries.

Reply:
Yes I can add this ability to this role, there is another thing I thought about that can also help us, its something I ran it to with ocp, for inventory hosts that need additional dns records like api.* there should be another inventory group used only for those additional dns records

Reply:
I have added the changes to be able to exclude certain ansible groups as described in the readme file